templates/registration/login.html

```diff
@@ -0,0 +1,47 @@
+{% extends "base.html" %}
+
+{% block content %}
+
+<div class="container">
+
+    <div class="row">
+        <div class="col-md-12">
+
+            {% if form.errors %}
+              <p>Your username and password didn't match. Please try again.</p>
+            {% endif %}
+
+            {% if next %}
+                {% if user.is_authenticated %}
+                    <p>You are already logged in!.</p>
+                {% else %}
+                    <p>Please login.</p>
+                {% endif %}
+            {% endif %}
+
+            <form method="post" action="{% url 'login' %}">
+            {% csrf_token %}
+
+            <div>
+                <td>{{ form.username.label_tag }}</td>
+                <td>{{ form.username }}</td>
+            </div>
+            <div>
+                <td>{{ form.password.label_tag }}</td>
+                <td>{{ form.password }}</td>
+            </div>
+
+            <div>
+              <input type="submit" value="login" class="btn"/>
+              <input type="hidden" name="next" value="{{ next }}" />
+            </div>
+            </form>
```

Comment:
```
<form...>
{% csrf_token %}
{{ form.as_p }}
<input type="submit" value="login" class="btn"/>
</form>
```

is probably fine. it'd fix the `<td>` in the middle of the page and add the field.errors that are missing

Reply:
I blame the mozilla docs!